qmd/sets.qmd

```diff
@@ -22,119 +22,351 @@ source("R/qmd-utils.R")
+      next()
+    }
+    
+    # Declare display elements
+    element_label <-
+      element_info[[parts_sheet_column_names$part_label_column_name]]
+    element_description <-
+      element_info[[parts_sheet_column_names$part_description_column_name]]
+    element_status <-
+      element_info[[parts_sheet_column_names$part_status_column_name]]
+    element_first_released <-
+      element_info[[parts_sheet_column_names$part_first_release_column_name]]
+    element_last_updated <-
+      element_info[[parts_sheet_column_names$part_last_updated_column_name]]
+    
+    # Verify display elements
+    verify_input(element_label,
+                 glue::glue('Part label is missing for {element_ID}'))
+    verify_input(element_description,
+                 glue::glue('Part description is missing for {element_ID}'))
+    verify_input(element_status,
+                 glue::glue('Part status is missing for {element_ID}'))
+    verify_input(
+      element_first_released,
+      glue::glue('Part first released is missing for {element_ID}')
+    )
+    verify_input(element_last_updated,
+                 glue::glue('Part last updates is missing for {element_ID}'))
+    
+    list_elements_list[[element_ID]] <-
+      glue::glue(
+        '**{element_label}** <a href=\"/parts.html#{element_ID}\">({element_ID})</a>. {element_description} </br>Status: {element_status}. First released: {element_first_released}. Last updated: {element_last_updated}.'
+      )
+  }
+  list_of_lists[[list_ID]] <-
+    create_html_list(list_display_info, list_elements_list)
+}
+
+# Create list for categorical displays
+categories_list <- list()
+categories_info_table <- sets_table[sets_table[[sets_sheet_column_names$part_set_type_column_name]] == constants$set_type_is_catSet, ]
+unique_category_set_IDs <- unique(categories_info_table[[sets_sheet_column_names$part_set_ID_column_name]])
+for (category_set_ID in unique_category_set_IDs) {
+  category_info <- parts_table[parts_table[[parts_sheet_column_names$part_ID_column_name]]==category_set_ID, ]
+  if (nrow(category_info) < 1) {
+    warning(glue::glue('{category_set_ID} has no valid info in parts and is skipped'))
+    next()
+  }
+  category_label <-
+    category_info[[parts_sheet_column_names$part_label_column_name]]
+  verify_input(set_type_label,
+               glue::glue('Category label is missing for {category_set_ID}'))
+  category_instruction <-
+    category_info[[parts_sheet_column_names$part_instruction_column_name]]
+  verify_input(
+    set_type_instruction,
+    glue::glue('Category instructions are missing for {category_set_ID}')
+  )
+  category_description <-
+    category_info[[parts_sheet_column_names$part_description_column_name]]
+  verify_input(set_type_description,
+               glue::glue('Category description is missing for {category_set_ID}'))
+  
+  category_display <- glue::glue(
+    '## {category_label} {{ #{category_set_ID} }}
```

Comment:
This template string is the same as on lines https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L233-L236 and https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/sets-lists-implementation/qmd/sets.qmd#L89-L91 and should be moved into a function.